app/Main.hs

```diff
@@ -157,16 +164,24 @@ showInput :: Input -> Text
 showInput Stdin = "(input)"
 showInput (File fp) = pack fp
 
+infixl 3 <??>
+(<??>) :: Maybe a -> a -> a
+(<??>) = flip fromMaybe
+
 -- | Utility function to validate and format swarm-lang code
-formatFile :: Input -> IO ()
-formatFile input = do
+formatFile :: Input -> Maybe Width -> IO ()
+formatFile input mWidth = do
   content <- getInput input
   case readTerm content of
-    Right t -> do
-      case t of
-        Nothing -> Text.putStrLn ""
-        Just ast -> Text.putStrLn $ prettyText ast
-      exitSuccess
+    Right Nothing -> Text.putStrLn ""
+    Right (Just ast) -> do
+      mWindow <- Term.size
+      let mkOpt w = LayoutOptions (AvailablePerLine w 1.0)
+      let opt =
+            fmap mkOpt mWidth
+              <|> fmap (\(Term.Window _h w) -> mkOpt w) mWindow
+                <??> defaultLayoutOptions
```

Comment:
If you use an `infixr` version of `<??>` / `?` you can just write this as a chain (untested):
```
fmap mkOpt mWidth
  ? fmap (\(Term.Window _h w) -> mkOpt w) mWindow
  ? defaultLayoutOptions
```

Reply:
Works great thanks!